src/sqlfluff/core/rules/std/L026.py

```diff
@@ -53,10 +53,10 @@ def _lint_references_and_aliases(
 
                 violation_buff.append(
                     LintResult(
-                        # Return the segment rather than the string
-                        anchor=tbl_ref[1],
+                        # Return the first segment rather than the string
+                        anchor=tbl_ref.segments[0],
```

Comment:
Also tidying, as with L025, but here, L026 references the `ObjectReferencePart` that was renamed, `segment` -> `segments`. So here, using the names rather than subscripts helps surface an error (what was previously a `BaseSegment` reference is now a `List[BaseSegment]`).